src/core/enc.rs

```diff
@@ -206,26 +206,16 @@ encode_ix!(
 impl Encode for types::Bytes<&'_ [u8]> {
     #[inline]
     fn encode<W: Write>(&self, writer: &mut W) -> Result<(), Error<W::Error>> {
-        TypeNum::new(major::BYTES << 5, self.0.len() as u64).encode(writer)?;
+        types::Bytes::bounded(self.0.len(), writer)?;
         writer.push(self.0)?;
         Ok(())
     }
 }
 
-pub struct BytesStart;
-
-impl Encode for BytesStart {
-    #[inline]
-    fn encode<W: Write>(&self, writer: &mut W) -> Result<(), Error<W::Error>> {
-        writer.push(&[(major::BYTES << 5) | marker::START])?;
-        Ok(())
-    }
-}
-
 impl Encode for &'_ str {
     #[inline]
     fn encode<W: Write>(&self, writer: &mut W) -> Result<(), Error<W::Error>> {
-        TypeNum::new(major::STRING << 5, self.len() as u64).encode(writer)?;
+        types::BadStr::bounded(self.len(), writer)?;
```

Comment:
maybe it should be called `UncheckedStr`

Reply:
Do you mean introducing an extra `UncheckedStr` or renaming `BadStr` to `UncheckedStr`?

Reply:
rename it